use_cluster.py

```diff
@@ -366,4 +355,415 @@ def return_cluster_dict(q_cluster, q_ids_list):
+        if(term_in_cluster): 
+            #If term in cluster, from NMI paper making sure that a frequent term in other clusters, but not in this gets negative NMI
+            #See page 3 of NMI paper, and reference [1] on footer
+            expected_percentage_term_occur_in_cluster = len(clus_qs)/len(lemmatized_qs_list)
+            actual_percentage_term_occur_in_cluster = occurences_term_in_cluster/(occurences_term_not_in_cluster + occurences_term_in_cluster)
+        I_t_c = 0.0
+        P_t0/=total_num_qs
+        P_t1/=total_num_qs
+        P_u0/=total_num_qs
+        P_u1/=total_num_qs
+        P_t0_u0/=total_num_qs
+        P_t0_u1/=total_num_qs
+        P_t1_u0/=total_num_qs
+        P_t1_u1/=total_num_qs
+        combined_P = {'00': P_t0_u0 ,'01': P_t0_u1 ,'10': P_t1_u0 ,'11': P_t1_u1}
+        single_T = {'0': P_t0 ,'1': P_t1}
+        single_U = {'0': P_u0 ,'1': P_u1}
+
+        for _ , (key, value) in enumerate(combined_P.items()):
+            if(value!=0):
+                I_t_c+= (value * math.log((value / (single_T[str(key[0])] * single_U[str(key[1])])),2))
+            else:
+                I_t_c+=0
+
+        H_t = 0.0
+        H_c = 0.0
+        for i in range(2):
+            if(single_T[str(i)]!=0):
+                H_t+=single_T[str(i)]* math.log(single_T[str(i)],2)
+            else:
+                H_t+=0
+            if(single_U[str(i)]!=0):
+                H_c+=single_U[str(i)]* math.log(single_U[str(i)],2)
+            else:
+                H_c+=0
+        H_t*=-1
+        H_c*=-1
+        if(I_t_c != 0 and (H_c + H_t) !=0):
+            NMI_i_t_c = 2*(I_t_c/(H_c + H_t))
+        else:
+            NMI_i_t_c = 0
+        if(term_in_cluster):
+            if(actual_percentage_term_occur_in_cluster < expected_percentage_term_occur_in_cluster):
+                logging.debug("Term where percentage of actual occurences are lesser than expected, if in cluster :", term)
+                logging.debug("Actual %:", actual_percentage_term_occur_in_cluster)
+                logging.debug("Expected %:", expected_percentage_term_occur_in_cluster)
+                NMI_i_t_c*=-1 #As from the paper, to ensure this term doesn't get selected
+        else:
+            NMI_i_t_c = 0
+        NMI_scores.append([term, NMI_i_t_c])
+    return NMI_scores
+        
+"""
+> The objective of this function is to:
+1. Calculate the New Score which is from the combination of the NMI score and the distances from Centroid
+    
+2. Takes in the NMI Scores and the Distances, and returns the list of New Scores according to the metric created
+
+3. Requires the Maximum NMI and the Global Minimum Distance for scaling and equal weighing as defined in metric
+
+:The inputs required are: NMI Scores, Distances, Maximum NMI Score and Global Minimum Distance
+:The format of NMI_Scores and Distances as is defined in calculate_label_score
+"""
+
+def compute_new_score(NMI_Scores, Distances, Max_NMI_Score, Global_Min_Distance):
```

Comment:
Can you give this a more descriptive name? Maybe compute_combined_nmi_score or compute_distance_nmi_score?

Reply:
Hey Turner, I agree! Have renamed it to compute_distance_nmi_score as that describes the crux of it well! :D